processor/src/decoder/mod.rs

```diff
@@ -444,7 +444,8 @@ impl Decoder {
     /// operation to the trace.
     pub fn start_call(&mut self, fn_hash: Word, addr: Felt) {
         // get the current clock cycle here (before the trace table is updated)
-        let clk = self.trace_len();
+        // clk is guaranteed never to exceed a u32 value, we can change the format here
```

Comment:
You are right. In general, `trace_len` should never exceed `clk`, right? So we should also change that in a future PR?

Reply:
Aside from removing the comment (which is accurate but unnecessary imo), I think everything is good here as it is. I'll elaborate a bit on how the decoder works to try to clarify and hopefully address your question (but let me know if I've misunderstood)

We're processing the MAST to figure out what we need to execute (and to tie those operations to the stack) and to prove that the execution we're doing matches the specified MAST (i.e. we're actually rebuilding the MAST as we go, so that the root at the end of the decoder trace matches the root specified as a public input). Therefore, every time we append execution of an operation to the end of the decoder operation (such as here, where we append a `CALL` operation), we know that the clock cycle of this operation is also equal to the current `trace_len`, since the current trace only contains the preceding operations that the decoder has already processed. However, they still represent conceptually different things, since the `trace_len` will change as soon as this op is appended but the `clk` cycle for this operation won't.

Please let me know if that doesn't answer your question or if I've misunderstood (and feel free to ping me if it's easier to chat through things directly)